server/version/version.go

```diff
@@ -14,20 +18,35 @@ type Server struct{}
 // Version returns the version of the API server
 func (s *Server) Version(context.Context, *empty.Empty) (*version.VersionMessage, error) {
 	vers := common.GetVersion()
-	ksonnetVersion, err := ksutil.KsonnetVersion()
+	ksonnetVersion, err := ksutil.Version()
+	if err != nil {
+		return nil, err
+	}
+	kustomizeVersion, err := kustomize.Version()
+	if err != nil {
+		return nil, err
+	}
+	helmVersion, err := helm.Version()
+	if err != nil {
+		return nil, err
+	}
+	kubectlVersion, err := kube.Version()
```

Comment:
I have some concerns that every time Version endpoint is called, we fork 4 processes. Not only is this inefficient for subsequent calls (since these values would never change), but since version is an unauthorized endpoint, there is potential for DoS-ing the API server.

I think we should mitigate this by this by preloading or memoizing all versions and caching it somewhere so that version is simply returning already calculated versions.

Reply:
Good point, I didn’t consider this. Will work on it.